### PR TITLE
trivy 0.24.2 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/trivy.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Trivy < Formula
+  desc "Vulnerbility scanner for container images, file systems, and Git repos"
+  homepage "https://aquasecurity.github.io/trivy/"
+  url "https://github.com/aquasecurity/trivy/archive/v0.24.2.tar.gz"
+  sha256 "ddbffec10e32de2dfbef14169028d60df7504ae255ac674febbafdcf61c7906b"
+  license "Apache-2.0"
+  head "https://github.com/aquasecurity/trivy.git", branch: "main"
+
+  depends_on "go" => :build
+  depends_on "go" => :build
```
Comment (cho-m) [inline Formula/trivy.rb:10]: Duplicate line

File: Formula/trivy.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X=main.version=#{version}"), "./cmd/trivy"
+  end
+
+  test do
+    output = shell_output("#{bin}/trivy image alpine:3.10")
+    assert_match "(UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 1)", output
```
Comment (carlocab) [inline Formula/trivy.rb:18]:
```suggestion
    output = shell_output("#{bin}/trivy image alpine:3.10")
    assert_match(/\(UNKNOWN: \d+, LOW: \d+, MEDIUM: \d+, HIGH: \d+, CRITICAL: \d+\)/, output)
```
These numbers can change at any time (presumably without changes to the `trivy` source) so let's just match with a regex.

File: Formula/trivy.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/trivy.rb:10]: 😆

File: Formula/trivy.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/trivy.rb:18]: Yeah, I think the most important piece of info in here is the `CRITICAL: 1`

File: Formula/trivy.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/trivy.rb]: I think at one moment, I was thinking to use `docker` for `test`, it turns out it is not needed.

File: Formula/trivy.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Trivy < Formula
+  desc "Vulnerbility scanner for container images, file systems, and Git repos"
```
Comment (SMillerDev) [inline Formula/trivy.rb:2]:
```suggestion
  desc "Vulnerability scanner for container images, file systems, and Git repos"
```

**Comment by knqyf263** [comment]
Hi, I'm a maintainer of Trivy. I'm just curious why you build a binary. We distribute binaries for macOS and Linux.
https://github.com/aquasecurity/trivy/releases/tag/v0.24.2

**Comment by carlocab** [comment]
All software packaged in this repository is built from source. The short answer to why is that we do it as a matter of policy. The longer answer can get quite long, but the reasons are similar to those of other package managers which do the same. (Most of them do.)

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**